modules/generator/registry/registry.go

```diff
@@ -156,7 +156,7 @@ func (r *ManagedRegistry) NewHistogram(name string, buckets []float64, histogram
 	// are disabled, eventually the new implementation can handle all cases
 
 	if hasNativeHistograms(histogramOverride) {
-		h = newNativeHistogram(name, buckets, r.onAddMetricSeries, r.onRemoveMetricSeries, traceIDLabelName, histogramOverride)
+		h = newNativeHistogram(name, buckets, r.onAddMetricSeries, r.onRemoveMetricSeries, traceIDLabelName, histogramOverride, r.externalLabels)
```

Comment:
adding external labels here is a new feature? or fixing a bug where we weren't applying external labels to native histos?

Reply:
The labels used to get passed on the `collectMetrics` function, and was just moved up to avoid the extra calls there, since they don't change after the histogram is created.